### PR TITLE
Add new zsh completion and update README.md

### PR DESCRIPTION
**Pull request by RODNOTGOD**

### CODE REVIEW

**Comment by primis** [comment]
Maybe make authors of that "pimterry et al."?

**Comment by RODNOTGOD** [comment]
sure thing

File: README.md [hunk 1]
```diff
@@ -29,20 +29,26 @@ curl https://cdn.rawgit.com/pimterry/notes/v0.2.0/notes > /usr/local/bin/notes &
...
 
+Bash
 ```bash
 curl https://cdn.rawgit.com/pimterry/notes/v0.2.0/notes.bash_completion > /usr/share/bash-completion/completions/notes
 ```
 
+Zsh
+```bash
+curl https://cdn.rawgit.com/pimterry/notes/v0.2.0/notes._notes > /usr/share/zsh/site-functions/_notes
+```
```
Comment (pimterry) [inline README.md]: We've kind of interleaved the two sets of setup here: you need bash or zsh, here's the bash dir here's the zsh dir, here's the bash commands here's the zsh commands. We should just split this into two sections.

File: _notes [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# Description
+# -----------
+#
+#  Simple delightful note taking, with more unix and less lock-in. (https://github.com/pimterry/notes)
+#
+# ------------------------------------------------------------------------------
+# Authors
+# -------
+#
+#  * pimterry et al.
```
Comment (pimterry) [inline _notes:12]: You're more than welcome to add a link to https://github.com/pimterry/notes/graphs/contributors, which would give people credit a little more directly :smile:.

File: _notes [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+{
+  local -a list
+  list=(
+    new:'Create new file'
+    ls:'<pattern> List notes by path'
+    find:'[pattern] Search notes by filename and path'
+    grep:'<pattern> Search notes by content'
+    open:'<name> Open a notes for editing by full name'
+    rm:'[-r | --recursive] <name> Remove note, or folder if -r or --recursive is given]'
+    --help:'Show usage'
```
Comment (pimterry) [inline _notes]: We now also have 'search' here, as of #16

File: _notes [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  local note_dir="${configured_dir:-$HOME/notes}"
+
+  if (($CURRENT == 2)); then
+    _alternative 'sub-commands:files:__notes_cmd' && _ret=0
+  elif (($CURRENT == 3)); then
+    case $words[2] in
+      open|rm)
+        _path_files -W "${note_dir}" && _ret=0;;
+      new|ls)
+        _path_files -W "${note_dir}" -/ && _ret=0;;
```
Comment (pimterry) [inline _notes]: Find and search should probably auto complete paths too?

File: _notes [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  fi
+}
+
+# Local Variables:
+# mode: Shell-Script
+# sh-indentation: 2
+# indent-tabs-mode: nil
+# sh-basic-offset: 2
+# End:
+# vim: ft=zsh sw=2 ts=2 et
```
Comment (pimterry) [inline _notes]: I'd rather not have these in here inline. I'd be quite open to a separate PR adding a [.editorconfig](http://editorconfig.org/) file or something though.

**Review by pimterry — CHANGES_REQUESTED** [review]
Thanks so much for doing this! I like zsh, but I've never written a Zsh completion script, so I was somewhat dreading having to try and wade through this, this is a great help. I've added notes for a few tweaks inline. 

I'm also interested in testing, which I know might not be what you want to hear. We did manage to get it working very nicely for the bash autocompletion though (https://github.com/pimterry/notes/blob/master/test/test-bash-completion.bats) which gives me some hope it might be doable, and at least getting a couple of tests in would be a huge boost to my confidence that we'll be able to keep this working reliably in future!

File: _notes (hunk 5 above)
Comment (RODNOTGOD) [reply _notes]: This portion of the comments were for the [zsh-completion](https://github.com/zsh-users/zsh-completions) repo. It was part of their guide lines, but this depends mostly if you wanted this to be added to the zsh-completions repo.

File: _notes (hunk 4 above)
Comment (RODNOTGOD) [reply _notes]: Alright sounds good

**Comment by RODNOTGOD** [comment]
I'll work on the tweaks. On the bats test I'm a little iffy on them; simply because bats doesn't play nicely with zsh and zsh completions don't behave life bash scripts and outputs.

**Comment by pimterry** [comment]
You don't need to close this PR, you can just push more commits to the same branch, and they'll magically appear here too :smile:.

That is a good argument against zsh completion testing, I must admit. If we could get even the basics to check that it hasn't been fundamentally broken in future that would be useful though. https://github.com/bamarni/symfony-console-autocomplete might be worth a look - it's a project including autocomplete for zsh and bash, with a basic set of bats tests for both that might be helpful, and which seem to work. Have a go, see what happens.

**Comment by pimterry** [comment]
@RODNOTGOD Any thoughts on this? I'd love to get this merged soon if possible.

**Comment by RODNOTGOD** [comment]
Hey sorry for the extended delay, college picked up again and I've been busy. Now regarding the auto-complete issue. It seems to be that there no effective way to correctly emulate the zsh tests for correct auto  completion. The symfony-autocomplete looked promising, but turns out only can test for zsh file generation. I looked at other testing methods, but most seem to implement also a file generation. So to conclusion, We can either push the zsh without testing; and zsh tends not to break anything, but does goes against what you see for the project. Or we push this to [zsh-completions](https://github.com/zsh-users/zsh-completions) repository, but creates a divergence in the project now D:.

**Comment by pimterry** [comment]
@RODNOTGOD Ok - I'm happy to ignore automated zsh testing for the moment then. Let's just finish up the rest of the comments left on here, and then it's ready to go.

**Comment by RODNOTGOD** [comment]
Ok, sounds good to me. I'll update the pull request by today or tomorrow.

**Review by pimterry — APPROVED** [review]

**Pull request merged**